### PR TITLE
Add skip to main content link

### PR DESCRIPTION
**Pull request by josephfusco**
This change implements a standard [skip navigation link](https://webaim.org/techniques/skipnav/) which allows users to skip over the navigation to avoid having to tab through it on every page.

Demo
https://www.loom.com/share/b5a25182478c4ec189372461accf0131

### CODE REVIEW

File: src/components/SkipNavigationLink/SkipNavigationLink.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+import styles from './SkipNavigationLink.module.scss';
+import * as SELECTOR from 'constants/selectors';
```
Comment (apmatthews) [inline src/components/SkipNavigationLink/SkipNavigationLink.js:1]: Nit pick: Would `SELECTORS` (plural) read more clearly? Or perhaps `import {MAIN_CONTENT_ID} from ...` directly?

File: src/components/Main/Main.js [hunk 2]
```diff
@@ -8,9 +10,14 @@ import styles from './Main.module.scss';
...
  * @returns {React.ReactElement} The Main component.
  */
-export default function Main({ children, className }) {
+export default function Main({ children, className, ...props }) {
   return (
-    <main className={[styles.main, className || undefined].join(' ')}>
+    <main
+      id={SELECTOR.MAIN_CONTENT_ID}
+      tabIndex="-1"
+      className={[styles.main, className || undefined].join(' ')}
```
Comment (theodesp) [inline src/components/Main/Main.js:18]: This does not need  the `undefined`

```suggestion
      className={[styles.main, className].join(' ')}
```

**Review by apmatthews — COMMENTED** [review]
Awesome work, Joe! One thing to consider is that skipping to `#main-content` also skips over the page/post title and author/date info. Could we possibly skip differently depending on whether there is content in the header?

File: src/components/SkipNavigationLink/SkipNavigationLink.js (hunk 1 above)
Comment (josephfusco) [reply src/components/SkipNavigationLink/SkipNavigationLink.js:1]: @apmatthews I would be in favor of updating to SELECTORS.

I feel `SELECTORS.MAIN_CONTENT_ID` reads better when there are many constants on a page and it helps visually namespace it, however seeing that there is only 1 right now I have no preference :)

File: src/components/Main/Main.js (hunk 2 above)
Comment (josephfusco) [reply src/components/Main/Main.js:18]: Thanks @theodesp will remove

**Comment by josephfusco** [comment]
> Awesome work, Joe! One thing to consider is that skipping to `#main-content` also skips over the page/post title and author/date info. Could we possibly skip differently depending on whether there is content in the header?

@apmatthews Yes excellent point. This may require some layout refactoring but thanks for catching that. Page & post titles should definitely be in the content.

File: src/components/SkipNavigationLink/SkipNavigationLink.js (hunk 1 above)
Comment (apmatthews) [reply src/components/SkipNavigationLink/SkipNavigationLink.js:1]: Works for me! I don't have a preference either but your reasoning for `SELECTORS` makes sense 👍🏻

File: src/components/index.js [hunk 3]
```diff
@@ -1,49 +1,53 @@
...
-import SearchResults from './SearchResults';
 import SearchRecommendations from './SearchRecommendations';
-import NextLinkWrapper from './NextLinkWrapper';
-import LoadingSearchResult from './LoadingSearchResult';
-import TaxonomyTerms from './TaxonomyTerms';
+import SearchResults from './SearchResults';
 import SEO from './SEO';
+import SkipNavigationLink from './SkipNavigationLink';
+import TaxonomyTerms from './TaxonomyTerms';
+import Testimonials from './Testimonials';
```
Comment (josephfusco) [inline src/components/index.js:26]: A-Z sort :)

**Review by apmatthews — APPROVED** [review]
This is great! Makes our document structure much better. Funny how things that might seem like they are "just for accessibility" end up making things better for all. 🚀

**Review by apmatthews — APPROVED** [review]

**Pull request merged**